frontend-react/src/shared/CodeSnippet/CodeSnippet.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { Icon, Tooltip } from "@trussworks/react-uswds";
+import classnames from "classnames";
+
+import styles from "./CodeSnippet.module.scss";
+
+interface CodeSnippetProps extends React.PropsWithChildren {
+    copyString: string;
+}
+
+export const CodeSnippet = ({ children, copyString }: CodeSnippetProps) => {
+    const [isCopied, setIsCopied] = useState(false);
+
+    /**
+     * Cached component that renders tooltip so that changing isCopied status
+     * causes it to remount (thus forcing tooltip position recalculation).
+     */
+    const CopyTooltip = useCallback(
+        ({ children }: React.PropsWithChildren) => (
+            <Tooltip
+                className="fixed-tooltip"
+                position="left"
+                label={isCopied ? "Copied" : "Copy to clipboard"}
+                onClick={() => {
+                    navigator.clipboard.writeText(copyString);
+                    setIsCopied(true);
+                }}
+            >
+                {children}
```

Comment:
You actually don't need `children` here anymore since you're just passing in the string which is all the content anyways. Currently you're repeating both the string and the content when using the component, you should only need to pass it in once.

Can we just use `copyString` in replace of `children`?

Reply:
I don't agree. The _presentation_ of the snippet and the _actual snippet_ are two entirely separate things that need to be given to our component. Until we have a workflow that creates the _presentation_ from an _actual snippet_ that both need to be provided manually.

Reply:
USWDS has plenty of components where you provide the text content via a prop. We shouldn't have to pass in the some content twice. And every `CodeSnippet` component has the same icon, and should, so we also shouldn't need to pass that in manually either.

Reply:
Apologies, we're talking about two different things. In regards to the tooltip content specifically: having it all embedded within the callback factory is fine.